src/main/java/com/nike/cerberus/service/SafeDepositBoxService.java

```diff
@@ -347,27 +355,21 @@ public SafeDepositBoxV2 updateSafeDepositBoxV2(final SafeDepositBoxV2 safeDeposi
      * @param id The unique identifier for the safe deposit box
      */
     @Transactional
-    public void deleteSafeDepositBox(final Set<String> groups, final String id) {
-        final Optional<SafeDepositBoxV2> box = getAssociatedSafeDepositBoxV2(groups, id);
-
-        if (!box.isPresent()) {
-            throw ApiException.newBuilder()
-                    .withApiErrors(DefaultApiError.ENTITY_NOT_FOUND)
-                    .build();
-        }
+    public void deleteSafeDepositBox(VaultAuthPrincipal vaultAuthPrincipal, final String id) {
+        final SafeDepositBoxV2 box = getSafeDepositBoxByIdAndValidatePrincipalAssociationV2(vaultAuthPrincipal, id);
```

Comment:
I like long method names but at some point they become harder to read.  How about the original method name "getAssociatedSafeDepositBoxV2" or some another briefer name "getSDBUsingPrincipal"?